rkt/fetch.go

```diff
@@ -111,45 +143,53 @@ func fetchImage(img string, ds *cas.Store, ks *keystore.Keystore, discover bool)
 			if _, ok := app.Labels["version"]; !ok {
 				latest = true
 			}
-			return fetchImageFromEndpoints(ep, ds, ks, latest)
+			return fetchImageFromEndpoints(ep, ascFile, ds, ks, latest)
 		}
 	}
-	if err != nil {
-		return "", fmt.Errorf("not a valid URL (%s)", img)
-	}
+
 	switch u.Scheme {
-	case "http", "https", "docker":
+	case "http", "https", "docker", "file":
 	default:
 		return "", fmt.Errorf("rkt only supports http, https or docker URLs (%s)", img)
 	}
-	return fetchImageFromURL(u.String(), u.Scheme, ds, ks, false)
+	return fetchImageFromURL(u.String(), u.Scheme, ascFile, ds, ks, false)
 }
 
-func fetchImageFromEndpoints(ep *discovery.Endpoints, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
-	return downloadImage(ep.ACIEndpoints[0].ACI, ep.ACIEndpoints[0].ASC, "", ds, ks, latest)
+func fetchImageFromEndpoints(ep *discovery.Endpoints, ascFile *os.File, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
+	return fetchImageFrom(ep.ACIEndpoints[0].ACI, ep.ACIEndpoints[0].ASC, "", ascFile, ds, ks, latest)
 }
 
-func fetchImageFromURL(imgurl string, scheme string, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
-	return downloadImage(imgurl, ascURLFromImgURL(imgurl), scheme, ds, ks, latest)
+func fetchImageFromURL(imgurl string, scheme string, ascFile *os.File, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
+	return fetchImageFrom(imgurl, ascURLFromImgURL(imgurl), scheme, ascFile, ds, ks, latest)
 }
 
-func downloadImage(aciURL string, ascURL string, scheme string, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
-	stdout("rkt: fetching image from %s", aciURL)
+func fetchImageFrom(aciURL string, ascURL string, scheme string, ascFile *os.File, ds *cas.Store, ks *keystore.Keystore, latest bool) (string, error) {
+	if scheme != "file" || globalFlags.Debug {
+		stdout("rkt: fetching image from %s", aciURL)
+	}
+
 	if globalFlags.InsecureSkipVerify {
-		stdout("rkt: warning: signature verification has been disabled")
+		if ks != nil {
```

Comment:
this is redundant, ks will be nil iff globalFlags.InsecureSkipVerify (maybe we should clean this up a bit)